### DISCUSSION TITLE
Advanced stop lines (aka bike boxes)

### DISCUSSION THREAD

**Discussion by fabiobettani** [streetcomplete/StreetComplete#5361] [Ideas]
I was wondering if StreetComplete can be used to help map advanced stop lines for bikes (https://wiki.openstreetmap.org/wiki/Tag:cycleway%3Dasl).

It would certainly be useful to check occurrences of "cycleway=asl" with a missing parameter "direction=*", and ask the user about what direction is served by the advanced stop line.

It would also be great to ask the user whether a given intersection has advanced stop lines or not. But I don't know if this is technically doable, because each ASL would need to be mapped as a separate point, in proximity to the intersection (so it's not a matter of adding a property to an existing object: it's about creating new objects). This quest should also be limited to intersections with traffic light.

**Comment by mnalis** [comment]
> It would certainly be useful to check occurrences of "cycleway=asl" with a missing parameter "direction=*", and ask the user about what direction is served by the advanced stop line.

I'm guessing yes. It should probably use similar quest type as ["Which direction leads upwards"](https://wiki.openstreetmap.org/wiki/StreetComplete/Quests#incline_direction) quest. Unless someone comes up here with reasons not to, you should open an new issue and [fill in quest suggestion template](https://github.com/streetcomplete/StreetComplete/issues/new?assignees=&labels=&projects=&template=quest-suggestion.md)  for that part of the idea!


> It would also be great to ask the user whether a given intersection has advanced stop lines or not. But I don't know if this is technically doable, because each ASL would need to be mapped as a separate point, in proximity to the intersection (so it's not a matter of adding a property to an existing object: it's about creating new objects). This quest should also be limited to intersections with traffic light.

Pristine StreetComplete does not allow things like that currently, but [SCEE](https://wiki.openstreetmap.org/wiki/SCEE) (StreetComplete "Expert Edition" fork) allows it like following:

- long press on the map on approximate location of ASL
- choose `Insert node into way`
- make sure correct way _and_ ASL position on it are selected (and confirm)
- type `asl` into preset search and click on `Advanced Stop Line`

(you should also make sure to enable `POI: Cycling` beforehand and set it to display `cycleway=asl`; or you wouldn't know where ASLs are already mapped. Note that this will of course clutter a display little more, and is not as easy as regular SC quest. But a quest for _adding new ASLs_ is unlikely to happen, both because it would be extremely spammy, and because there is no tag to map there is no ASL)

**Reply by matkoniecz** [reply]
note that in many cases manual tagging of `direction` is a pointless waste of time, and it should be asked only when dubious at all (close to crossings in both directions and allowing two-way traffic)

**Reply by mnalis** [reply]
Yes, the [wiki](https://wiki.openstreetmap.org/wiki/Tag:cycleway%3Dasl) mentions it is not needed in `oneway` case, so the quest should be excluded on those. 
If there is a consensus that `direction=*` is useless in many other cases too, perhaps that clarification should be added to the wiki? 
As currently written, wiki seems to highly recommend (if not straight out _require_) adding `direction` tag in all cases except `oneway`, and  [over 62%](https://taginfo.openstreetmap.org/tags/cycleway%3Dasl#combinations) of `cycleway=asl` do have `direction=*` already tagged at the moment. 
Also, [id-tagging-schema for ASL](https://github.com/openstreetmap/id-tagging-schema/blob/main/data/presets/cycleway/asl.json) seems to define `direction` in preferred `fields` (and not more optional `moreFields`)

**Comment by matkoniecz** [comment]
> It would also be great to ask the user whether a given intersection has advanced stop lines or not. But I don't know if this is technically doable, because each ASL would need to be mapped as a separate point, in proximity to the intersection (so it's not a matter of adding a property to an existing object: it's about creating new objects). This quest should also be limited to intersections with traffic light.

the tricky part is guessing where ASL are, asking for every question is bad idea, asking for every crossing with signals is a bad idea

also, do we have tag for "no ASL on this crossing"?

**Reply by matkoniecz** [reply]
hmm, maybe it can be displayable on cycleway overlay like parking overlay allows to add street narrowing?
